streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -788,9 +793,9 @@ protected StreamTask createStreamTask(TaskId id, Collection<TopicPartition> part
 
         sensors.taskCreationSensor.record();
 
-        ProcessorTopology topology = builder.build(id.topicGroupId);
-
-        return new StreamTask(id, applicationId, partitions, topology, consumer, producer, restoreConsumer, config, sensors, stateDirectory, cache);
+        final ProcessorTopology topology = builder.build(id.topicGroupId);
+        final RecordCollector recordCollector = new RecordCollectorImpl(producer, id.toString());
+        return new StreamTask(id, applicationId, partitions, topology, consumer, restoreConsumer, config, sensors, stateDirectory, cache, recordCollector);
```

Comment:
Sounds good.

About the high-coupling thing: yeah I definitely agree. If there are any step-by-step refactoring that could be done earlier let's do that.